spec/plugins/data_table/server_side.js

```diff
@@ -480,54 +481,184 @@ describe("DataTable Plugin", function() {
+      });
+    });
+
+    it("should close the activeFilterMenu when the user clicks to open another menu, and then open the new menu", function() {
+      var cg = table.configGenerator();
+      var lastFilterMenu;
+      var currentFilterMenu;
+      table.dataTable.find("thead th").each(function () {
+        var wrapper = $(".DataTables_sort_wrapper", this);
+        if (wrapper) {
+          var title = wrapper.text();
+          var col = cg.columnConfigByTitle.attributes[title];
+          if (col && col.filter) {
+            if (currentFilterMenu) {
+              lastFilterMenu = currentFilterMenu;
+            }
+            currentFilterMenu = $(".filterMenu", this);
+            expect(currentFilterMenu.is(":hidden"));
+            $("span", this).trigger("click");
+            expect(currentFilterMenu.is(":visible"));
+            if (lastFilterMenu) {
+              expect(lastFilterMenu.is(":hidden"));
+            }
           }
-          else if (col.filter.type === "list") {
-            // test assignment
-            $('.filterMenu #value', this).prop("checked", true).trigger("change");
-            expect(col.filter.value).toEqual(["Basic", "Advanced"]);
-            // verify ajax
-            expectedFilterObj = [{type:"list", attr: col.attr, comparison:"value", value:["Basic", "Advanced"]}];
-            VerifyFilterAjax(expectedFilterObj);
-
-            // test unassignment
-            $('.filterMenu #value', this).prop("checked", false).trigger("change");
-            expect(col.filter.value).toEqual(null);
-            // verify ajax
-            expectedFilterObj = [];
-            VerifyFilterAjax(expectedFilterObj);
+        }
+      });
+    });
+
+    it("should clear the filters when the clear button is clicked", function() {
+      var cg = table.configGenerator();
+      table.dataTable.find("thead th").each(function () {
+        var wrapper = $(".DataTables_sort_wrapper", this);
+        if (wrapper) {
+          var title = wrapper.text();
+          var col = cg.columnConfigByTitle.attributes[title];
+          if (col && col.filter) {
+            if (col.filter.type === "string") {
+              $(".filterMenu input", this).val("Test").trigger("change");
+              $(".btn-clear", this).click();
+              expect($(".filterMenu input", this).val()).toEqual("");
+            }
+            else if (col.filter.type === "numeric") {
+              $(".filterMenu #eq", this).val("3").trigger("change");
+              $(".btn-clear", this).click();
+              expect($(".filterMenu #eq", this).val()).toEqual("");
+            }
+            else if (col.filter.type === "list") {
+              $(".filterMenu #value", this).prop("checked", true).trigger("change");
+              $(".btn-clear", this).click();
+              $(".filterMenu #value", this).prop("checked", false).trigger("change");
+              expect($(".filterMenu #value", this).prop("checked")).toEqual(false);
+            }
```

Comment:
I think making different test data would be best, then I can test each column manually.